AGENTS.md

```diff
@@ -0,0 +1,88 @@
+# Metal3 Ironic Image - AI Agent Instructions
+
+Instructions for AI coding agents. For project overview, see [README.md](README.md).
+For contribution guidelines, see [CONTRIBUTING.md](CONTRIBUTING.md).
+
+## Repository Structure
+
+| Directory | Purpose |
+|-----------|---------|
+| `scripts/` | Entry point scripts (runironic, rundnsmasq, runhttpd, etc.) |
+| `ironic-config/` | Jinja2 templates for Ironic/dnsmasq/httpd configuration |
+| `hack/` | CI scripts (shellcheck, markdownlint) |
+
+## Key Files
+
+| File | Purpose |
+|------|---------|
+| `Dockerfile` | Main image build definition |
+| `prepare-image.sh` | Image setup (Ironic, dnsmasq, httpd configs) |
+| `configure-nonroot.sh` | Non-root user setup for security |
+| `ironic-packages-list` | Python packages (Ironic, IPA, dependencies) |
+| `upper-constraints.txt` | Python version pins (from OpenStack release) |
+
+## Testing Standards
+
+CI uses GitHub Actions (`.github/workflows/`). Run locally before PRs:
+
+| Command | Purpose |
+|---------|---------|
+| `make build` | Build container image |
+| `./hack/shellcheck.sh` | Shell script linting |
+| `./hack/markdownlint.sh` | Markdown linting |
+
+## Code Conventions
+
+- **Shell**: Use `set -euxo pipefail` in scripts
+- **Jinja2**: Templates in `ironic-config/` render at container runtime
+
+## Key Workflows
+
+### Modifying Entry Point Scripts
+
+1. Edit script in `scripts/`
+1. Run `./hack/shellcheck.sh`
+1. Test with `make build` and manual container run
+
+### Modifying Configuration Templates
+
+1. Edit Jinja2 template in `ironic-config/`
+1. Document new environment variables in `README.md`
+1. Test configuration renders correctly at runtime
+
+### Updating Ironic Version
+
+1. Update `ironic-packages-list` with new version
+1. Update `upper-constraints.txt` (from OpenStack release)
```

Comment:
Would it be worth it to add the URL of the OpenStack requirements repository as source for this file ?